//
// * Copyright 2019 The Feast Authors
// *
// * Licensed under the Apache License, Version 2.0 (the "License");
// * you may not use this file except in compliance with the License.
// * You may obtain a copy of the License at
// *
// *     https://www.apache.org/licenses/LICENSE-2.0
// *
// * Unless required by applicable law or agreed to in writing, software
// * distributed under the License is distributed on an "AS IS" BASIS,
// * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// * See the License for the specific language governing permissions and
// * limitations under the License.
//

syntax = "proto3";
package feast.core;

option java_package = "feast.proto.core";
option java_outer_classname = "SourceProto";
option go_package = "github.com/feast-dev/feast/sdk/go/protos/feast/core";


message Source {

    // The kind of data source Feast should connect to in order to retrieve FeatureRow value
    SourceType type = 1;

    // Source specific configuration
    oneof source_config {
        KafkaSourceConfig kafka_source_config = 2;
    }
}

enum SourceType {
    INVALID = 0;
    KAFKA = 1;
}

message KafkaSourceConfig {
    // Comma separated list of Kafka bootstrap servers. Used for feature sets without a defined source host[:port]]
    string bootstrap_servers = 1;

    // Kafka topic to use for feature sets without user defined topics
    string topic = 2;

    // Number of Kafka partitions to to use for managed feature stream.
    int32 partitions = 3;

    // Defines the number of copies of managed feature stream Kafka.
    int32 replicationFactor = 4;
}